plugins/gatsby-plugin-generate-json/gatsby-node.js

```diff
@@ -0,0 +1,24 @@
+const fs = require('fs');
+const path = require('path');
+
+exports.onPostBuild = async ({ graphql, store }, pluginOptions) => {
+  const { program } = store.getState();
```

Comment:
what does this store.getState() do? little confused here!

Reply:
Gatsby uses a redux store [under the hood](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/redux/index.ts) to handle information about the in environment / Gatsby site. We're using `program` from the store so that we can get access path for the `public` build directory.

One of the plugins we use did the same thing, so we decided to follow a similar approach.

Reply:
gotcha, that makes sense! thank you